Documentation/ApiOverview/Authentication/CSRFlikeRequestTokenHandling.rst

```diff
@@ -0,0 +1,163 @@
+..  include:: /Includes.rst.txt
+..  index:: Authentication, Request token
+..  _authentication-request-token:
+
+================================
+CSRF-like request token handling
+================================
+
+..  versionadded:: 12.0
+
+A CSRF-like request token handling is available to mitigate potential cross-site
+requests on actions with side effects. This approach does not require an
+existing server-side user session, but uses a nonce (number used once) as a
+"pre-session". The main scope is to ensure a user actually has visited a page,
```

Comment:
I don't understand it, can we link somewhere or explain pre-sessions and nonce?

Reply:
I understand pre-session as something that must be valid (request token) prior login. So you have to visit the page before posting the credentials. That's why it is in quotes.

Reply:
Nonce is linked to the almighty Wikipedia :)